### PR TITLE
[go_router_builder] add support to go_router_builder for initializing a StatefulShellBranch with observers

### PR DESCRIPTION
**Pull request by satoyan**
This pr added the ability to provide a list of NavigatorObservers to a StatefulShellBranch. 
The equivalent was never added to go_router_builder, so this pr provides that functionality.

With this change, you can provide NavigatorObservers directly to the StatefulShellBranch and fix https://github.com/flutter/flutter/issues/143869.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by satoyan** [comment]
or anyone else create PR instead of me, please! :pray:

**Comment by hangyujin** [comment]
Hi @satoyan , thank you for your contribution, can you also add tests to this pr?

**Comment by satoyan** [comment]
Thank you for your replying! I added some test case and update version to pass CI.
Please tell me anything else I need to do :bow: 
thank you.
@hangyujin

**Comment by hangyujin** [comment]
@johnpryan  Hi John, can you take a  second look at this?

**Review by johnpryan — APPROVED** [review]

File: packages/go_router_builder/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
```
Comment (hangyujin) [inline packages/go_router_builder/CHANGELOG.md:1]: update to 2.5.0

Comment (satoyan) [reply packages/go_router_builder/CHANGELOG.md]: Thank you!  I fixed that. :smile:

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by chunhtai** [comment]
looks like there is a conflict and ci failing, can you fix them?

**Review by chunhtai — COMMENTED** [review]
Looks like ci expecting fluter >= 3.13.0. can you update it

**Pull request merged**